### PR TITLE
test: add real model tests for nullable enum null guards

### PR DESCRIPTION
**Pull request by galesky-a**
**Description**
* Adds a set of tests to verify null handling when calling `.Equals` and `.Hasheable` on unknown enums

### CODE REVIEW

File: Adyen.Test/StringEnumConverterTest.cs [hunk 1]
```diff
@@ -52,4 +52,118 @@ public void Given_StringEnumConverter_When_Value1_Returns_Value1_As_Enum()
...
+        {
+            // Use a real model class (Modification) to verify null guards work
+            var json = @"{""status"": ""someUnknownFutureStatus""}";
+            var a = JsonConvert.DeserializeObject<Adyen.Model.TransferWebhooks.Modification>(json);
+            var b = JsonConvert.DeserializeObject<Adyen.Model.TransferWebhooks.Modification>(json);
+            
+            Assert.IsNotNull(a);
+            Assert.IsNull(a.Status, "Unknown enum value should deserialize to null");
+            Assert.IsTrue(a.Equals(b), "Equals should not throw NRE when enum is null");
+        }
```
[the remark this answers is not in the document]
Comment (galesky-a) [reply Adyen.Test/StringEnumConverterTest.cs:152]: I've added the test case you suggested comparing a null Status against a known Status value.

However, calling `.Equals()` on a `Nullable<T>` where the value is null does **not** throw a `NullReferenceException`. `StatusEnum?` is `Nullable<StatusEnum>` — a value type (struct), not a reference type. When `HasValue` is false, `.Equals()` simply returns `false` without throwing.
